### PR TITLE
Updates to CLI Reference - RHDEVDOCS-1521

### PR DESCRIPTION
**Pull request by nkakkar81**
Following changes have been made as part of this PR:
1. CLI updates as per OCP Guidelines
2. Added missing/new commands w.r.t latest Pipelines release 
3. Added examples

### CODE REVIEW

**Comment by nkakkar81** [comment]
@Preeticp - raised a PR for CLI changes. Request you to please review. Thanks

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Delete a clustertask resource in a cluster.
+
+.Example: Delete `mytask1` and `mytask2` clustertasks
+----
+$ tkn clustertask delete mytask1 mytask2
+----
+
+
+== tkn clustertask describe
+Describes a clustertask.
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: 'Describe' to keep it consistent with list, manage, etc

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+.Example: Describe the `mytask` clustertask
+----
+$ tkn clustertask describe mytask1
+----
+
+== tkn clustertask list
+List clustertasks in a namespace.
+
+.Example: List clustertasks in a namespace
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: ... in the `myspace` namespace?

File: modules/ROOT/pages/op-tkn-cli-pipeline-management.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+== tkn pipeline list
+List pipelines in a namespace.
+
+.Example: Display a list of pipelines
+-----
+$ tkn pipeline list
+-----
+
+== tkn pipeline logs
+Display pipeline logs for a specific pipeline .
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-pipeline-management.adoc]: nit-extra space before the period

File: modules/ROOT/pages/op-tkn-cli-pipeline-management.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+.Example: Stream live logs for the `mypipeline` pipeline
+----
+$ tkn pipeline logs -f mypipeline
+----
+
+== tkn pipeline start
+Start a pipeline.
+
+.Example: Start a pipeline
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-pipeline-management.adoc]: Start `mypipeline`

File: modules/ROOT/pages/op-tkn-cli-pipeline-run.adoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
+// Module included in the following assemblies:
+//
+// * pipelines/op-tkn-cli-references.adoc
+
+[id="cli-pipeline-run-commands_{context}"]
+= Pipelines Run Commands
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-pipeline-run.adoc]: Pipelinerun Commands?

File: modules/ROOT/pages/op-tkn-cli-task-run.adoc [hunk 6]
```diff
@@ -0,0 +1,56 @@
+// Module included in the following assemblies:
+//
+// *  pipelines/op-tkn-cli-reference.adoc
+
+[id="cli-task-run-commands_{context}"]
+= Task  Run Commands
+
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-task-run.adoc]: Taskrun

File: modules/ROOT/pages/op-tkn-cli-task-management.adoc [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+.Example: Create a task defined by the `mytask.yaml` file in a namespace
+----
+$ tkn task create -f mytask.yaml -n myspace
+----
+
+== tkn task delete
+Delete task resources in a namespace.
+
+.Example: Delete tasks from a namespace
+----
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-task-management.adoc]: Delete `mytask1` and `mytask2`... to keep it consistent with taskrun etc.

File: modules/ROOT/pages/op-tkn-cli-utility-commands.adoc [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+.Example: Completion code for `bash` shell
+----
+$ tkn completion bash
+----
+
+== tkn version
+Print version information of the Tekton CLI `tkn`.
+
+.Example: Check for the newer version availability
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-utility-commands.adoc]: ...availability of a newer version

**Comment by Preeticp** [comment]
Looks great.
Some minor things and it is ready to be merged.

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc (hunk 2 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: Have tried not to use `myspace` namespace throughout. I think we discussed this before.

**Comment by nkakkar81** [comment]
@Preeticp - Have incorporated all comments, except one. Please review

File: modules/ROOT/nav.adoc [hunk 9]
```diff
@@ -21,5 +21,15 @@
 ** xref:proc_interacting-with-pipelines-using-the-developer-perspective.adoc[4.2 Interacting with pipelines using the Developer perspective]
 * xref:assembly_cli-reference.adoc[5. CLI Reference]
 ** xref:proc_installing-cli.adoc[5.1 Installing CLI]
-** xref:ref_cli-reference.adoc[5.2 Basic Commands]
+** xref:op-tkn-cli-reference.adoc[5.2 Basic Commands]
+*** xref:op-tkn-cli-utility-commands.adoc[5.2.1 Utility Commands]
```
Comment (Preeticp) [inline modules/ROOT/nav.adoc]: In view of pushing these changes to OCP repo, and making them consistent with those docs, can you please change the topic names to:

op-tkn-cli-reference> tkn CLI Reference
- op-installing tkn> Installing tkn
- op-tkn-basic-commands> Basic tkn commands

Comment (nkakkar81) [reply modules/ROOT/nav.adoc]: @Preeticp  - Thank you for your comments. I have incorporated the comments, and have also made changes to the headings of each module to be in sentence case like in OCP Docs  here - https://docs.openshift.com/container-platform/4.3/cli_reference/openshift_cli/administrator-cli-commands.html

Please review, and let me know if there are any more changes required.

File: modules/ROOT/nav.adoc [hunk 10]
```diff
@@ -20,6 +20,16 @@
 ** xref:proc_creating-applications-with-cicd-pipelines.adoc[4.1 Creating-applications with CI/CD pipelines]
 ** xref:proc_interacting-with-pipelines-using-the-developer-perspective.adoc[4.2 Interacting with pipelines using the Developer perspective]
 * xref:assembly_cli-reference.adoc[5. CLI Reference]
-** xref:proc_installing-cli.adoc[5.1 Installing CLI]
-** xref:ref_cli-reference.adoc[5.2 Basic Commands]
+** xref:proc_installing-cli.adoc[5.1 Installing tkn]
+** xref:op-tkn-cli-reference.adoc[5.2 Basic tkn commands]
+*** xref:op-tkn-cli-utility-commands.adoc[5.2.1 Utility commands]
```
Comment (Preeticp) [inline modules/ROOT/nav.adoc]: We could probably remove either tkn/cli from the title. Wdyt?

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc [hunk 11]
```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// *  pipelines/op-tkn-cli-reference.adoc
+
+[id="cli-clustertask-management-commands_{context}"]
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: I try to keep the IDs same as the file names, easy to copy paste and also if you need to use ifdef it helps to be consistent. But I prefer the shortened version. As mentioned above we could either keep tkn/cli.

File: modules/ROOT/pages/op-tkn-cli-reference.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
+[id='pipeline-cli-reference']
+= Pipeline CLI reference
+//include::modules/openshift-developer-cli-attributes.adoc[]
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-reference.adoc]: please remove this we will have our pipeline attributes

File: modules/ROOT/pages/assembly_cli-reference.adoc [hunk 13]
```diff
@@ -5,15 +5,15 @@
 
 :context: cli-reference
 
-With the Tekton Pipelines command-line interface (CLI) you can interact with Tekton from a terminal. This section describes how to install it and lists the basic commands. 
+With the Tekton Pipelines command-line interface (CLI) you can interact with Tekton from a terminal. This section describes how to install it and lists the basic commands.
```
Comment (Preeticp) [inline modules/ROOT/pages/assembly_cli-reference.adoc]: with Tekton (tkn)

File: modules/ROOT/pages/op-tkn-cli-reference.adoc [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+//include::modules/common-attributes.adoc[]
+:context: pipeline-cli-reference
+toc::[]
+
+
+This section lists basic Tekton CLI commands. For a more detailed description, see link:https://github.com/tektoncd/cli[Tekton Pipelines CLI].
+
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-reference.adoc]: I think we have more details now in our docs, do we want to still send the user to the upstream docs?

File: modules/ROOT/pages/op-tkn-cli-trigger-management.adoc [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+Manage trigger templates.
+
+.Example: Display trigger template help
+----
+$ tkn triggertemplate -h
+----
+== tkn triggertemplate delete
+Delete trigger template in a namespace.
+
+.Example: Delete `mytemplate1` and `mytemplate2` trigger template in a namespace
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-cli-trigger-management.adoc]: templates, L64 and 66

**Comment by Preeticp** [comment]
@nkakkar81 looks great. Some minor comments and a few suggestions.

File: modules/ROOT/pages/op-tkn-cli-reference.adoc (hunk 14 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-cli-reference.adoc]: Right now we are listing only the command name along with an example. We are not explaining the various parameters and values for any of the commands. Therefore, in case user wants to know about the commands in detail, they can refer to tkn cli.

File: modules/ROOT/nav.adoc (hunk 10 above)
Comment (nkakkar81) [reply modules/ROOT/nav.adoc]: Changed the file names to remove 'cli'

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc (hunk 11 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: done

File: modules/ROOT/pages/op-tkn-cli-reference.adoc (hunk 12 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-cli-reference.adoc]: done

File: modules/ROOT/pages/assembly_cli-reference.adoc (hunk 13 above)
Comment (nkakkar81) [reply modules/ROOT/pages/assembly_cli-reference.adoc]: done

File: modules/ROOT/pages/op-tkn-cli-trigger-management.adoc (hunk 15 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-cli-trigger-management.adoc]: done

**Comment by nkakkar81** [comment]
@Preeticp  - All comments incorporated. Please review. Thanks

File: modules/ROOT/pages/assembly_cli-reference.adoc [hunk 16]
```diff
@@ -5,15 +5,15 @@
 
 :context: cli-reference
```
Comment (Preeticp) [inline modules/ROOT/pages/assembly_cli-reference.adoc:6]: Do we want to change the title to op-tkn-cli... or do you want to do that in a separate PR?

File: modules/ROOT/pages/op-tkn-clustertask-management.adoc [hunk 17]
```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// *  pipelines/op-tkn-reference.adoc
+
+[id="tkn-clustertask-management-commands_{context}"]
+= Clustertask management commands
```
Comment (Preeticp) [inline modules/ROOT/pages/op-tkn-clustertask-management.adoc:6]: It is best practice and easier to maintain the same names for file names and ID/context. For example in this case it should be op-tkn-...management

File: modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc (hunk 11 above)
Comment (Preeticp) [reply modules/ROOT/pages/op-tkn-cli-clustertask-management.adoc]: This is not the same as of now, please see my comment.

**Comment by Preeticp** [comment]
@nkakkar81 if you decide to take care of all the ID/context and file name changes in another PR, this LGTM, and I will go ahead and merge this PR.

File: modules/ROOT/pages/assembly_cli-reference.adoc (hunk 16 above)
Comment (nkakkar81) [reply modules/ROOT/pages/assembly_cli-reference.adoc:6]: I think we can do this as part of moving the assembly to OCP

File: modules/ROOT/pages/op-tkn-clustertask-management.adoc (hunk 17 above)
Comment (nkakkar81) [reply modules/ROOT/pages/op-tkn-clustertask-management.adoc:6]: Ahh...I thought op-tkn should not be part of the id.  I will modify all the files.

**Comment by nkakkar81** [comment]
@Preeticp - the ID/Context is now aligned with the file names. Please review and merge if you don't have further comments. Thanks!!

File: modules/ROOT/pages/assembly_cli-reference.adoc [hunk 18]
```diff
@@ -5,15 +5,15 @@
...
+With the Tekton (tkn) Pipelines command-line interface (CLI) you can interact with Tekton from a terminal. This section describes how to install it and lists the basic commands.
 
 
 // CLI Installation Guide
 include::proc_installing-cli.adoc[leveloffset=+1]
 
 
 // CLI Reference
-include::ref_cli-reference.adoc[leveloffset=+1]
+include::op-tkn-cli-reference.adoc[leveloffset=+1]
```
Comment (Preeticp) [inline modules/ROOT/pages/assembly_cli-reference.adoc]: op-tkn-reference?

Comment (nkakkar81) [reply modules/ROOT/pages/assembly_cli-reference.adoc]: done with the change.
